workflow-ui/container-android/src/main/java/com/squareup/workflow1/ui/modal/ModalContainer.kt

```diff
@@ -75,6 +75,13 @@ public abstract class ModalContainer<ModalRenderingT : Any> @JvmOverloads constr
               dialogView,
               findParentLifecycle = { parentLifecycleOwner?.lifecycle }
             )
+            // Ensure that each dialog has its own ViewTreeSavedStateRegistryOwner,
+            // so views in each dialog layer don't clash with other layers.
+            ref.savedStateRegistryKey = Compatible.keyFor(newScreen, i.toString())
```

Comment:
`buildDialog` is an abstract method. By passing the id through it we're relying on subclasses to handle it properly. I did it this way to have one less mistake they could make. WDYT?

Reply:
Gotcha. I guess its a bit awkward either way. Anyway - this is fine enough :)

Reply:
FWIW:

```kotlin
          // This is the unique id we'll use to make stateRegistryAggregator put a
          // SavedStateRegistryOwner on the dialog's decorView, as required by Compose.
          // We put it in this sketchy lateInit field on DialogRef rather than passing
          // it through abstract buildDialog method so that subclasses can't screw it up.
          ref.savedStateRegistryKey = Compatible.keyFor(modal, i.toString())
```